be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1042,7 +1045,7 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
     }
 
     if (!_is_need_vec_eval && !_is_need_short_eval) {
-        _output_non_pred_columns(block);
+        _output_columns_by_relace(block, _first_read_column_ids);
```

Comment:
non pred columns != first read columns. If there is a delete condition, the first read columns will contains delete condition columns.